xmodule/assets/video/public/js/03_video_player.js

```diff
@@ -0,0 +1,911 @@
+}
+
+// ***************************************************************
+// Public functions start here.
+// These are available via the 'state' object. Their context ('this'
+// keyword) is the 'state' object. The magic private function that makes
+// them available and sets up their context is makeFunctionsPublic().
+// ***************************************************************
+
+function destroy() {
+    let player = this.videoPlayer.player;
+    this.el.removeClass([
+        'is-unstarted', 'is-playing', 'is-paused', 'is-buffered',
+        'is-ended', 'is-cued'
+    ].join(' '));
+    $(window).off('.video');
+    this.el.trigger('destroy');
+    this.el.off();
+    this.videoPlayer.stopTimer();
+    if (this.resizer && this.resizer.destroy) {
+        this.resizer.destroy();
+    }
+    if (player && player.video) {
+        player.video.removeEventListener('loadedmetadata', this.videoPlayer.onLoadMetadataHtml5, false);
+    }
+    if (player && _.isFunction(player.destroy)) {
+        player.destroy();
+    }
+    if (this.canPlayHLS && player.hls) {
+        player.hls.destroy();
+    }
+    delete this.videoPlayer;
+}
+
+function pause() {
+    if (this.videoPlayer.player.pauseVideo) {
+        this.videoPlayer.player.pauseVideo();
+    }
+}
+
+function play() {
+    if (this.videoPlayer.player.playVideo) {
+        if (this.videoPlayer.isEnded()) {
+            // When the video will start playing again from the start, the
+            // start-time and end-time will come back into effect.
+            this.videoPlayer.goToStartTime = true;
+        }
+
+        this.videoPlayer.player.playVideo();
+    }
+}
+
+// This function gets the video's current play position in time
+// (currentTime) and its duration.
+// It is called at a regular interval when the video is playing.
+function update(time) {
+    this.videoPlayer.currentTime = time || this.videoPlayer.player.getCurrentTime();
+
+    if (isFinite(this.videoPlayer.currentTime)) {
+        this.videoPlayer.updatePlayTime(this.videoPlayer.currentTime);
+
+        // We need to pause the video if current time is smaller (or equal)
+        // than end-time. Also, we must make sure that this is only done
+        // once per video playing from start to end.
+        if (
+            this.videoPlayer.endTime !== null
+    && this.videoPlayer.endTime <= this.videoPlayer.currentTime
+        ) {
+            this.videoPlayer.pause();
+
+            this.trigger('videoProgressSlider.notifyThroughHandleEnd', {
+                end: true
+            });
+
+            this.el.trigger('stop');
+        }
+        this.el.trigger('timeupdate', [this.videoPlayer.currentTime]);
+    }
+}
+
+function setPlaybackRate(newSpeed) {
+    this.videoPlayer.player.setPlaybackRate(newSpeed);
+}
+
+function onSpeedChange(newSpeed) {
+    let time = this.videoPlayer.currentTime;
+
+    if (this.isFlashMode()) {
+        this.videoPlayer.currentTime = Time.convert(
```

Comment:
This 'Time' object does not exist we need to  import it, this was failing during karma JS testing. 
I tried to fix the test by this [import](https://github.com/openedx/edx-platform/blob/9d71ce8151de5323c298e921b6d6c1ee7cbf72a6/xmodule/assets/video/public/js/04_video_control.js#L3) but its better if we can update this here.